Avoid PHP warnings about undefined array keys when sanitizing malformed PEMs

crocodele opened the pull request:
Target branch: ~`4.0.x`~ `3.4.x`

- [X] It is a Bug fix
- [ ] It is a New feature
- [ ] It is related to dependencies

Includes:
- [ ] Breaks BC
- [ ] Deprecations

When given a malformed PEM string, `sanitizePEM()` triggers two PHP warnings: `Undefined array key 0` and `Undefined array key 1`.

Spomky commented:
Hi,

Many thanks for this PR.
The target branch should be 3.4.x.
Also, the result of `preg_match_all` should be verified in a similar way to [line 400](https://github.com/web-token/jwt-framework/blob/3.4.x/src/Library/KeyManagement/KeyConverter/KeyConverter.php#L400)

crocodele committed:
Avoid PHP warnings about undefined indexes when sanitizing malformed PEMs

crocodele commented:
> Hi,
> 
> Many thanks for this PR. The target branch should be 3.4.x. Also, the result of `preg_match_all` should be verified in a similar way to [line 400](https://github.com/web-token/jwt-framework/blob/3.4.x/src/Library/KeyManagement/KeyConverter/KeyConverter.php#L400)

Switched the target branch to `3.4.x` and changed to checking the result of `preg_match_all()` and throwing an exception instead of null coalescing `$matches[0][0]` and `$matches[0][1]`, in https://github.com/web-token/jwt-framework/pull/556/commits/af17363edb37dda3dfa7ca034a6db21df9f1b722.

Spomky committed:
Add test for loading invalid PEM key

A new test function was added in ECKeysTest.php to handle cases of loading invalid PEM keys. An InvalidArgumentException is expected to be thrown with a specific error message when an invalid private PEM key is loaded.

Spomky commented:
Perfect. Thanks 👍🏼
I added a test to cover this part.
